src/components/BookDetail/BookDetail.js

```diff
@@ -1,17 +1,33 @@
 import React from 'react';
+import { Redirect } from 'react-router-dom';
 
-import books from '../../assets/books.json';
+import AxiosInstance from '../../config/AxiosInstance';
 
 import BookInformation from './BookInformation';
 import BookSuggestions from './BookSuggestions';
 import BookComments from './BookComments';
 
 class BookDetail extends React.Component {
-  state = {
-    book: books.find(book => book.id === parseInt(this.props.match.params.id, 10)) // eslint-disable-line react/prop-types
+  state = {};
+
+  componentDidMount = () => {
+    AxiosInstance.get(`/books/${this.props.match.params.id}`) // eslint-disable-line react/prop-types
```

Comment:
Move the interfaces against external services to a service file. i.e: BookService